src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -41,6 +42,7 @@ public Command parseCommand(String userInput) throws ParseException {
         }
 
         final String commandWord = matcher.group("commandWord");
+        final String flag = matcher.group("flag"); // returns a whitespace, followed by the tag e.g. " -f".
         final String arguments = matcher.group("arguments");
```

Comment:
In addition, I would like to request you to please parse arguments into a proper format other than a string (HashMap, maybe?) - or else this logic of parsing arguments will have to be repeated in all CommandParser classes. If you would prefer it done in a separate PR/by me, lmk as well.